field.go

```diff
@@ -207,6 +208,9 @@ func newField(path, index, name string, opts FieldOption) (*Field, error) {
 
 		remoteAvailableShards: roaring.NewBitmap(),
 
+		shardValidator: func(uint64) bool {
```

Comment:
what do you think about defining a `defaultShardValidator` somewhere and using that in place of this default code in 3 places?